discovery_engine/lib/src/ffi/types/init_config.dart

```diff
@@ -0,0 +1,113 @@
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import 'dart:ffi' show Pointer;
+
+import 'package:equatable/equatable.dart';
+import 'package:xayn_discovery_engine/src/domain/models/configuration.dart';
+import 'package:xayn_discovery_engine/src/domain/models/feed_market.dart';
+import 'package:xayn_discovery_engine/src/ffi/genesis.ffigen.dart'
+    show RustInitConfig;
+import 'package:xayn_discovery_engine/src/ffi/load_lib.dart' show ffi;
+import 'package:xayn_discovery_engine/src/ffi/types/feed_market_vec.dart'
+    show FeedMarketSliceFfi;
+import 'package:xayn_discovery_engine/src/ffi/types/string.dart' show StringFfi;
+import 'package:xayn_discovery_engine/src/infrastructure/assets/native/data_provider.dart';
+
+class InitConfigFfi with EquatableMixin {
+  final String apiKey;
+  final String apiBaseUrl;
+  final List<FeedMarket> feedMarkets;
+  final String smbertVocab;
+  final String smbertModel;
+  final String kpeVocab;
+  final String kpeModel;
+  final String kpeCnn;
+  final String kpeClassifier;
+
+  @override
+  List<Object?> get props => [
+        apiKey,
+        apiBaseUrl,
+        feedMarkets,
+        smbertVocab,
+        smbertModel,
+        kpeVocab,
+        kpeModel,
+        kpeCnn,
+        kpeClassifier,
+      ];
+
+  factory InitConfigFfi(
+    Configuration configuration,
+    NativeSetupData setupData,
+  ) =>
+      InitConfigFfi.fromParts(
+        apiKey: configuration.apiKey,
+        apiBaseUrl: configuration.apiBaseUrl,
+        feedMarkets: configuration.feedMarkets.toList(),
+        smbertVocab: setupData.smbertVocab,
+        smbertModel: setupData.smbertModel,
+        kpeVocab: setupData.kpeVocab,
+        kpeModel: setupData.kpeModel,
+        kpeCnn: setupData.kpeCnn,
+        kpeClassifier: setupData.kpeClassifier,
+      );
+
+  InitConfigFfi.fromParts({
+    required this.apiKey,
+    required this.apiBaseUrl,
+    required this.feedMarkets,
+    required this.smbertVocab,
+    required this.smbertModel,
+    required this.kpeVocab,
+    required this.kpeModel,
+    required this.kpeCnn,
+    required this.kpeClassifier,
+  });
+
+  void writeNative(Pointer<RustInitConfig> place) {
+    apiKey.writeNative(ffi.init_config_place_of_api_key(place));
+    apiBaseUrl.writeNative(ffi.init_config_place_of_api_base_url(place));
+    feedMarkets.toList().writeVec(ffi.init_config_place_of_markets(place));
+    smbertVocab.writeNative(ffi.init_config_place_of_smbert_vocab(place));
+    smbertModel.writeNative(ffi.init_config_place_of_smbert_model(place));
+    kpeVocab.writeNative(ffi.init_config_place_of_kpe_vocab(place));
+    kpeModel.writeNative(ffi.init_config_place_of_kpe_model(place));
+    kpeCnn.writeNative(ffi.init_config_place_of_kpe_cnn(place));
+    kpeClassifier.writeNative(ffi.init_config_place_of_kpe_classifier(place));
+  }
+
+  // used for testing only
```

Comment:
maybe for later, we could use annotations like `@visibleForTesting` from the [meta package](https://pub.dev/documentation/meta/latest/meta/visibleForTesting-constant.html) to let this be checked by the analysis tool.